### PR TITLE
Move dispatcher config

### PR DESCRIPTION
**Pull request by Liuhaai**
Related: #2694

### CODE REVIEW

File: dispatcher/dispatcher_test.go [hunk 1]
```diff
@@ -16,26 +16,29 @@ import (
...
 	"google.golang.org/protobuf/proto"
 
-	"github.com/iotexproject/iotex-core/config"
 	"github.com/iotexproject/iotex-proto/golang/iotexrpc"
 	"github.com/iotexproject/iotex-proto/golang/iotextypes"
 	"github.com/iotexproject/iotex-proto/golang/testingpb"
 )
 
+var (
+	defaultChainID uint32 = 1
```
Comment (dustinxie) [inline dispatcher/dispatcher_test.go:25]: this should be defined in chain.Default (when moving struct Chain {} out of config)

File: dispatcher/dispatcher_test.go [hunk 2]
```diff
@@ -16,26 +16,29 @@ import (
...
+		ProcessSyncRequestInterval: 0 * time.Second})
 	assert.NoError(t, err)
 	dp.AddSubscriber(chainID, &dummySubscriber{})
 	return dp
 }
 
 func startDispatcher(t *testing.T) (ctx context.Context, d Dispatcher) {
 	ctx = context.Background()
-	d = createDispatcher(t, config.Default.Chain.ID)
+	d = createDispatcher(t, defaultChainID)
```
Comment (dustinxie) [inline dispatcher/dispatcher_test.go:41]: let's revert this for now, see above comment

File: dispatcher/dispatcher_test.go (hunk 1 above)
Comment (dustinxie) [reply dispatcher/dispatcher_test.go:25]: also, let's move this part: https://github.com/iotexproject/iotex-core/blob/master/config/config.go#L77-L88
into p2p, define a p2p.Default for it

Comment (dustinxie) [reply dispatcher/dispatcher_test.go:25]: and also move https://github.com/iotexproject/iotex-core/blob/master/config/config.go#L77-L88
into p2p, define a p2p.Default for it

File: dispatcher/dispatcher.go [hunk 3]
```diff
@@ -19,14 +19,24 @@ import (
...
 	"github.com/iotexproject/iotex-core/pkg/lifecycle"
 	"github.com/iotexproject/iotex-core/pkg/log"
 	goproto "github.com/iotexproject/iotex-proto/golang"
 	"github.com/iotexproject/iotex-proto/golang/iotexrpc"
 	"github.com/iotexproject/iotex-proto/golang/iotextypes"
 )
 
+type (
+	// DispatcherConfig is the dispatcher config
+	DispatcherConfig struct {
```
Comment (CoderZhi) [inline dispatcher/dispatcher.go:31]: rename to Config

**Review by CoderZhi — APPROVED** [review]

File: dispatcher/dispatcher_test.go (hunk 1 above)
Comment (Liuhaai) [reply dispatcher/dispatcher_test.go:25]: > his should be defined in chain.Default (when moving struct Chain {} out of config

Should config.Default move out of config.go?
In #2695, the network default config is kept in config.go

Comment (dustinxie) [reply dispatcher/dispatcher_test.go:25]: config.Default stays in config.go, just become like 
```
Default = Config{
    Network: p2p.Default,
    Dispatcher: dispatcher.Default
}
```

**Review by dustinxie — APPROVED** [review]

**Pull request merged**